sdk/search/Azure.Search/src/Models/IndexDocumentsAction{T}.cs

```diff
@@ -0,0 +1,216 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Diagnostics;
+using System.Text.Json;
+using Azure.Core;
+
+#pragma warning disable SA1402 // File may only contain a single type
+
+namespace Azure.Search.Models
+{
+    // Hide the untyped IndexAction
+    [CodeGenSchema("IndexAction")]
+    internal partial class IndexAction { }
+
+    /// <summary>
+    /// Represents an index action that operates on a document.
+    /// </summary>
+    /// <typeparam name="T">
+    /// The .NET type that maps to the index schema. Instances of this type can
+    /// be retrieved as documents from the index.
+    /// </typeparam>
+    public class IndexDocumentsAction<T> : IUtf8JsonSerializable
+    {
+        /// <summary>
+        /// The operation to perform on a document in an indexing batch.
+        /// </summary>
+        public IndexActionType ActionType { get; private set; }
+
+        /// <summary>
+        /// The document to index.
+        /// </summary>
+        public T Document { get; private set; }
+
+        /// <summary>
+        /// Private constructor to prevent direct instantiation.
+        /// </summary>
+        private IndexDocumentsAction() { }
+
+        /// <summary>
+        /// Initializes a new instance of the IndexDocumentsAction class.
+        /// </summary>
+        /// <param name="type">
+        /// The operation to perform on the document.
+        /// </param>
+        /// <param name="doc">The document to index.</param>
+        public IndexDocumentsAction(IndexActionType type, T doc)
+        {
+            Debug.Assert(Enum.IsDefined(typeof(IndexActionType), type));
+
+            ActionType = type;
+            Document = doc;
+        }
+
+        /// <summary>
+        /// Serialize the document write action.
+        /// </summary>
+        /// <param name="writer">The JSON writer.</param>
+        void IUtf8JsonSerializable.Write(Utf8JsonWriter writer)
+        {
+            Debug.Assert(writer != null);
+
+            writer.WriteStartObject();
+
+            writer.WritePropertyName("@search.action");
+            writer.WriteStringValue(ActionType.ToSerialString());
+
+            // TODO: XXXXX - Investigate a more efficient way of injecting properties into user models with System.Text.Json
```

Comment:
Was a bug number supposed to go in there?

Reply:
I'm going to pull up Alex's tool and replace all of these with tracking bugs in the PR feedback follow up.  It put these in as an easily searchable reminder to myself to file bugs.